include/LeMonADE/feature/FeatureNNInteractionBcc.h

```diff
@@ -0,0 +1,602 @@
+  //re-initializes the lattice and overwrites what FeatureExcludedVolumeSc has written.
+  //FeatureAttributes needs to be in front, because when a monomer is added to the system
+  //by a MoveAddScMonomer, its attribute has to be set before it is written to the lattice.
+  typedef LOKI_TYPELIST_2(
+      FeatureAttributes,
+      FeatureExcludedVolumeBcc<FeatureLatticeType<lattice_value_type> >) 
+    required_features_front;
+
+
+  //! check for all Monte Carlo moves without special check functions (always true)
+  template<class IngredientsType>
+    bool checkMove(const IngredientsType& ingredients,const MoveBase& move) const;
+
+  //! check move for bcc-BFM local move. always throws std::runtime_error
+  template<class IngredientsType>
+    bool checkMove(const IngredientsType& ingredients,const MoveLocalSc& move) const;
+
+  //! check move for bcc-BFM local move
+  template<class IngredientsType>
+    bool checkMove(const IngredientsType& ingredients,MoveLocalBcc& move) const;
+
+  //! apply function for all Monte Carlo moves without special apply functions (does nothing)
+  template<class IngredientsType>
+    void applyMove(const IngredientsType& ing, const MoveBase& move){}
+
+  //! apply function for sc-BFM local move (always throws std::runtime_error)
+  template<class IngredientsType>
+    void applyMove(const IngredientsType& ing, const MoveLocalSc& move);
+
+  //! apply function for adding a monomer in bcc-BFM
+  template<class IngredientsType>
+    void applyMove(IngredientsType& ing, const MoveAddBccMonomer& move);
+
+  //note: apply function for bcc-BFM local move is not necessary, because 
+  //job of moving lattice entries is done by the underlying FeatureLatticeType
+
+  //! guarantees that the lattice is properly occupied with monomer attributes
+  template<class IngredientsType>
+  void synchronize(IngredientsType& ingredients);
+
+  //!adds interaction energy between two types of monomers
+  void setNNInteraction(int32_t typeA,int32_t typeB,double energy);
+
+  //!returns the interaction energy between two types of monomers
+  double getNNInteraction(int32_t typeA,int32_t typeB) const;
+
+  //!export bfm-file read command !nn_interaction
+  template <class IngredientsType>
+  void exportRead(FileImport <IngredientsType>& fileReader);
+
+  //!export bfm-file write command !nn_interaction
+  template <class IngredientsType>
+  void exportWrite(AnalyzerWriteBfmFile <IngredientsType>& fileWriter) const;
+
+};
+
+
+//////////////////  IMPLEMENTATION OF MEMBERS //////////////////////////////////
+
+/**
+ * @brief Constructor
+ **/
+template<template<typename> class LatticeClassType>
+FeatureNNInteractionBcc<LatticeClassType>::FeatureNNInteractionBcc()
+{
+  //initialize the energy and probability lookups with default values
+  for(size_t n=0;n<256;n++)
+    {
+      for(size_t m=0;m<256;m++)
+        {
+	  interactionTable[m][n]=0.0;
+	  probabilityLookup[m][n]=1.0;
+        }
+    }
+}
+
```

Comment:
This could be wrapped up using std::vector instead of c-arrays, i guess, like
`FeatureNNInteractionBcc<LatticeClassType>::FeatureNNInteractionBcc():interactionTable(256,std::vector<double>(264,0.0)),probabilityLookup(256,std::vector<double>(264,1.0)){}`